packages/good-design/src/advanced/customswitch/CustomSwitch.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useCallback } from 'react'
+import { Pressable, Icon, View, Box } from 'native-base'
+import { first } from 'lodash'
+
+import SelectBox from './SelectBox'
+// import SwitchIcon from ' ../../assets/svg/arrow-swap.svg'
+
+const SwitchIcon = () => (
+  <g strokeLinecap="round" strokeLinejoin="round" strokeWidth="1.5">
+    <path d="m5 10.0909h14l-5.8396-5.0909" stroke="#B0B4BB"/>
+    <path d="m19 13.9091h-14l5.8396 5.0909" stroke="#B0B4BB"/>
+  </g>
+)
+
+const SelectListItem = (
+  {
+    chain,
+    onPress,
+    isListItem,
+    isListOpen,
+    isLeft,
+    listNumber,
+  }: {
+    chain: string,
+    onPress: ((isLeft: boolean) => void) | ((isLeft: boolean, listNumber: number) => void)
+    isListItem: boolean,
+    isListOpen: boolean,
+    isLeft: boolean,
+    listNumber: number
+    
+  }) => {
+  const type = isListItem ? "list" : "button"
+  const onItemPress = useCallback(() => onPress(isLeft, listNumber), [chain, onPress]);
```

Comment:
```suggestion
  const onItemPress = useCallback(() => onPress(chain), [chain, onPress]);
```

Reply:
again I've asked to pass **chain** to callback 
**DO NOT RELY ON INDEXES IF YOU HAVE UNIQUE IDS**